tests/unit/test_request_acquisition.py

```diff
@@ -71,10 +77,28 @@ def test_saver(pipeline):
 def test_canceller(pipeline):
     req_storage = storage.MemoryRequestsStorage()
     add_uid_assigner(pipeline)
+
     add_requests_saver(pipeline, req_storage)
     add_requests_canceller(pipeline, req_storage)
+
+    requests_cancelled = []
+    add_cancelled_requests_notifier(pipeline, requests_cancelled)
```

Comment:
This is a bit unfortunate, I have worked on a remote computer and forgot to push. Therefore, I will get back to this PR later.